In [1]:
import pandas as pd

In [2]:
df = pd.read_parquet("../data/df_end_transaction.parquet")
df_ma = pd.read_parquet("../data/df_end_transaction_ma.parquet")

In [3]:
df

product  price    amount       place            datetime
0       Целебные травы   3.90  1.242125  Анор Лондо 2216-01-02 16:28:00
2       Целебные травы   3.90  0.882450  Анор Лондо 2216-01-02 03:03:00
3       Целебные травы   3.90  0.621377  Анор Лондо 2216-01-02 03:12:00
4       Целебные травы   3.90  1.367161  Анор Лондо 2216-01-02 05:33:00
5       Целебные травы   3.90  1.988265  Анор Лондо 2216-01-02 13:57:00
...                ...    ...       ...         ...                 ...
875031           Эстус   9.28  1.079104   Фалькония 2218-09-27 14:50:00
875032           Эстус   9.28  1.580617   Фалькония 2218-09-27 06:16:00
875033           Эстус   9.28  1.738492   Фалькония 2218-09-27 11:14:00
875034           Эстус   9.28  0.430340   Фалькония 2218-09-27 22:24:00
875035           Эстус   9.28  0.090248   Фалькония 2218-09-27 17:50:00

[699477 rows x 5 columns]

(price-cost)*amount -> max <=> (price -> max) && (amount -> max)
price ∈ [concurrent_price, concurrent_price*1.2]